js/geo/lng_lat.js

```diff
@@ -82,9 +82,11 @@ LngLat.prototype.toString = function () {
 LngLat.convert = function (input) {
     if (input instanceof LngLat) {
         return input;
-    }
-    if (Array.isArray(input)) {
+    } else if (input && input.lng && input.lat) {
```

Comment:
This check is dodgy: `0, 0` is a valid point but this will fail since `0` is falsy